### PR TITLE
idb 60 Bugfixes / Preparations for Open Source Hardware

### PR DESCRIPTION
**Pull request by itspngu**
* Preparation of README.md for open sourcing of design files
* Fixed Indicator LEDs being stuck in ON state
* Renamed function for ESC LED to reflect the fact that it's for layer indication / next rev PCB has extra LED slot on RSFT FN
* Reverted Eager Debounce to Default

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by tzarc** [comment]
This can't be accepted, as the Lufa submodule has been removed. Please fix this up.

**Comment by itspngu** [comment]
> This can't be accepted, as the Lufa submodule has been removed. Please fix this up.

Sorry, I'm not very familiar with the inner workings of QMK, but I do not see any reference to Lufa in the changed files. Could you give me any pointers to what needs to be done? Thanks.

**Comment by tzarc** [comment]
I was referring to this in the diff:
image

You'll have to re-add the submodule, or revert your commit which removes it.

**Comment by itspngu** [comment]
I remember this now. It's the same thing all over again, Lufa caused me git troubles before. I'll figure out what we did to fix it last time, sorry.

File: keyboards/idb/idb_60/keymaps/pngu/keymap.c [hunk 1]
```diff
@@ -6,6 +6,8 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
     KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_NO,
     KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Z,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
     LT(2, KC_CAPS), KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+    KC_LSFT, KC_NUBS, KC_Y,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),
+    KC_LCTL, KC_LGUI, KC_LALT,                                     KC_SPC,                             KC_RALT, KC_MENU, KC_RGUI
     KC_LSFT, KC_NUBS, KC_Y,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_NO,
     KC_LCTL, KC_LGUI, KC_LALT,                                     KC_SPC,                             KC_RALT, KC_MENU, MO(1)
```
Comment (zvecr) [inline keyboards/idb/idb_60/keymaps/pngu/keymap.c:12]:
```suggestion
```

**Comment by itspngu** [comment]
I'm not sure if the extern inline stuff in idb_60.c is deprecated or if it needs to be updated from "esc" to "fn", sorry for being a headache

**Comment by zvecr** [comment]
Given the implementation is in the header, and that header is being included, removal of the functions in idb_60.c is best.

**Comment by itspngu** [comment]
> Given the implementation is in the header, and that header is being included, removal of the functions in idb_60.c is best.

Okay, I think I've made all suggested changes then, hope I didn't miss anything.

File: keyboards/idb/idb_60/idb_60.h [hunk 2]
```diff
@@ -3,18 +3,18 @@
...
 }
 
 inline void _idb_60_caps_led_off(void) {
     writePinLow(C5);
 }
 
-inline void _idb_60_esc_led_off(void) {
+inline void _idb_60_fn_led_off(void) {
     writePinLow(C4);
 }
```
Comment (zvecr) [inline keyboards/idb/idb_60/idb_60.h:20]: Maybe some backwards compatibility here, depends how many boards are in the wild with keymaps out of the repo?
```suggestion

#define _idb_60_esc_led_on _idb_60_fn_led_on
#define _idb_60_esc_led_off _idb_60_fn_led_off

```

**Comment by itspngu** [comment]
> Otherwise LGTM

Yeah, that's a good idea, thanks.

**Comment by itspngu** [comment]
Hey uh, just making sure I don't need to make any more changes to get this merged :)

**Comment by zvecr** [comment]
Thanks!

**Comment by Wilba6582** [comment]
@zvecr  I get these build errors now, all keymaps:

```Linking: .build/idb_idb_60_default.elf                                                              [ERRORS]
 |
 | c:/msys64/home/wilba/qmk_utils/avr8-gnu-toolchain/bin/../lib/gcc/avr/8.3.0/../../../../avr/bin/ld.exe: .build/obj_idb_idb_60_default/keyboards/idb/idb_60/keymaps/default/keymap.o: in function `layer_state_set_user':
 | D:\dev\qmk_firmware/keyboards/idb/idb_60/keymaps/default/keymap.c:40: undefined reference to `_idb_60_fn_led_on'
 | c:/msys64/home/wilba/qmk_utils/avr8-gnu-toolchain/bin/../lib/gcc/avr/8.3.0/../../../../avr/bin/ld.exe: D:\dev\qmk_firmware/keyboards/idb/idb_60/keymaps/default/keymap.c:42: undefined reference to `_idb_60_fn_led_off'
 | c:/msys64/home/wilba/qmk_utils/avr8-gnu-toolchain/bin/../lib/gcc/avr/8.3.0/../../../../avr/bin/ld.exe: .build/obj_idb_idb_60_default/keyboards/idb/idb_60/idb_60.o: in function `led_set_kb':
 | D:\dev\qmk_firmware/keyboards/idb/idb_60/idb_60.c:11: undefined reference to `_idb_60_caps_led_on'
 | c:/msys64/home/wilba/qmk_utils/avr8-gnu-toolchain/bin/../lib/gcc/avr/8.3.0/../../../../avr/bin/ld.exe: D:\dev\qmk_firmware/keyboards/idb/idb_60/idb_60.c:13: undefined reference to `_idb_60_caps_led_off'
 | collect2.exe: error: ld returned 1 exit status
 |
```

No idea why this doesn't build for me...

**Comment by yiancar** [comment]
@zvecr also does not work for me. 

`Linking: .build/idb_idb_60_via.elf                                                                  [ERRORS]
 | 
 | /usr/local/opt/avr-binutils/bin/avr-ld: /var/tmp//ccBVZhpQ.ltrans0.ltrans.o: in function `layer_state_set':
 | /Users/yiangosyiangou/Documents/AVR_Projects/qmk_firmware/keyboards/idb/idb_60/keymaps/via/keymap.c:40: undefined reference to `_idb_60_fn_led_on'
 | /usr/local/opt/avr-binutils/bin/avr-ld: /Users/yiangosyiangou/Documents/AVR_Projects/qmk_firmware/keyboards/idb/idb_60/keymaps/via/keymap.c:42: undefined reference to `_idb_60_fn_led_off'
 | /usr/local/opt/avr-binutils/bin/avr-ld: /var/tmp//ccBVZhpQ.ltrans0.ltrans.o: in function `led_set':
 | /Users/yiangosyiangou/Documents/AVR_Projects/qmk_firmware/keyboards/idb/idb_60/idb_60.c:11: undefined reference to `_idb_60_caps_led_on'
 | /usr/local/opt/avr-binutils/bin/avr-ld: /Users/yiangosyiangou/Documents/AVR_Projects/qmk_firmware/keyboards/idb/idb_60/idb_60.c:13: undefined reference to `_idb_60_caps_led_off'
 | collect2: error: ld returned 1 exit status
 | 
make[1]: *** [.build/idb_idb_60_via.elf] Error 1
make: *** [idb/idb_60:via] Error 1
Make finished with errors
`

**Pull request merged**